aws-crt-http/HttpConnection.cs

```diff
@@ -1,8 +1,245 @@
+    internal delegate void OnStreamCompleteNative(int errorCode);
+
+    public sealed class HttpRequestOptions
+    {
+        public string Method { get; set; }
+        public Uri Uri { get; set; }
+        public HttpHeader[] Headers { get; set; }
+        public OnStreamOutgoingBody OnStreamOutgoingBody { get; set; }
+        public OnStreamComplete OnStreamComplete { get; set; }
+        public OnIncomingHeaders OnIncomingHeaders { get; set; }
+        public OnIncomingHeaderBlockDone OnIncomingHeaderBlockDone { get; set; }
+        public OnIncomingBody OnIncomingBody { get; set; }
+    }
+
+    [StructLayout(LayoutKind.Sequential, CharSet=CharSet.Ansi)]
+    public struct HttpHeader
+    {
+        [MarshalAs(UnmanagedType.LPStr)]
+        public string Name;
+
+        [MarshalAs(UnmanagedType.LPStr)]
+        public string Value;
+
+        public HttpHeader(string name, string value)
+        {
+            Name = name;
+            Value = value;
+        }
+    }
+
+    public abstract class HttpStream
+    {
+        [SecuritySafeCritical]
+        internal static class API
+        {
+            static LibraryHandle library = new LibraryHandle();
+            public delegate Handle aws_dotnet_http_stream_new(
+                                    IntPtr connection,
+                                    [MarshalAs(UnmanagedType.LPStr)] string method,
+                                    [MarshalAs(UnmanagedType.LPStr)] string uri,
+                                    [In] HttpHeader[] headers,
+                                    UInt32 header_count,
+                                    OnStreamOutgoingBodyNative onStreamOutgoingBody,
+                                    OnIncomingHeadersNative onIncomingHeaders,
+                                    OnIncomingHeaderBlockDoneNative onIncomingHeaderBlockDone,
+                                    OnIncomingBodyNative onIncomingBody,
+                                    OnStreamCompleteNative onStreamComplete);
+            public delegate void aws_dotnet_http_stream_destroy(IntPtr stream);
+            public delegate void aws_dotnet_http_stream_update_window(IntPtr stream, UInt64 incrementSize);
+
+            public static aws_dotnet_http_stream_new make_new = NativeAPI.Bind<aws_dotnet_http_stream_new>();
+            public static aws_dotnet_http_stream_destroy destroy = NativeAPI.Bind<aws_dotnet_http_stream_destroy>();
+            public static aws_dotnet_http_stream_update_window update_window = NativeAPI.Bind<aws_dotnet_http_stream_update_window>();
+        }
+
+        internal class Handle : CRT.Handle
+        {
+            protected override bool ReleaseHandle()
+            {
+                API.destroy(handle);
+                return true;
+            }
+        }
+
+        internal Handle NativeHandle { get; private set; }
+
+        public HttpClientConnection Connection { get; protected set; }
+        public int ResponseStatusCode { get; protected set; }

```

Comment:
ResponseStatusCode is specific to client/request streams
It wouldn't exist on server/response streams

Reply:
Fixed